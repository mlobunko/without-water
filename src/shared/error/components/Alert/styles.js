import styled, { css } from 'styled-components';

export const AlertStyled = styled.div`
  ${props =>
    props.centerPage &&
    css`
      display: flex;
      flex-direction: column;
      justify-content: center;
      align-items: center;
      height: 60vh;
    `};
`;
