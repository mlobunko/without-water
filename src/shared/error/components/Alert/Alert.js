import React from 'react';
import PropTypes from 'prop-types';

import { Alert as AlertStrap } from 'reactstrap';
import { AlertStyled } from './styles';

export const Alert = ({ color, textError, centerPage }) => (
  <AlertStyled centerPage={centerPage}>
    <AlertStrap color={color}>{textError}</AlertStrap>
  </AlertStyled>
);

export default Alert;

Alert.propTypes = {
  color: PropTypes.string.isRequired,
  textError: PropTypes.string.isRequired,
  centerPage: PropTypes.bool
};
