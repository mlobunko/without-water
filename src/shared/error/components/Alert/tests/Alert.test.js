import React from 'react';
import { shallow } from 'enzyme';

import { Alert } from '../Alert';

describe('Alert component', () => {
  it('renders properly centerPage true', () => {
    const props = {
      color: 'dark',
      textError: 'error',
      centerPage: true
    };
    const wrapper = shallow(<Alert {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('renders properly centerPage false', () => {
    const props = {
      color: 'dark',
      textError: 'error',
      centerPage: false
    };
    const wrapper = shallow(<Alert {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
