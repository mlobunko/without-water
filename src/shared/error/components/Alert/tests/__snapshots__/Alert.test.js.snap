// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert component renders properly centerPage false 1`] = `
<styled.div
  centerPage={false}
>
  <Alert
    closeAriaLabel="Close"
    color="dark"
    fade={true}
    isOpen={true}
    tag="div"
    transition={
      Object {
        "appear": true,
        "baseClass": "fade",
        "baseClassActive": "show",
        "enter": true,
        "exit": true,
        "in": true,
        "mountOnEnter": false,
        "onEnter": [Function],
        "onEntered": [Function],
        "onEntering": [Function],
        "onExit": [Function],
        "onExited": [Function],
        "onExiting": [Function],
        "tag": "div",
        "timeout": 150,
        "unmountOnExit": true,
      }
    }
  >
    error
  </Alert>
</styled.div>
`;

exports[`Alert component renders properly centerPage true 1`] = `
<styled.div
  centerPage={true}
>
  <Alert
    closeAriaLabel="Close"
    color="dark"
    fade={true}
    isOpen={true}
    tag="div"
    transition={
      Object {
        "appear": true,
        "baseClass": "fade",
        "baseClassActive": "show",
        "enter": true,
        "exit": true,
        "in": true,
        "mountOnEnter": false,
        "onEnter": [Function],
        "onEntered": [Function],
        "onEntering": [Function],
        "onExit": [Function],
        "onExited": [Function],
        "onExiting": [Function],
        "tag": "div",
        "timeout": 150,
        "unmountOnExit": true,
      }
    }
  >
    error
  </Alert>
</styled.div>
`;
